lib/src/algorithm.dart

```diff
@@ -3,10 +3,10 @@ part of encrypt;
 /// Interface for the Algorithms.
 abstract class Algorithm {
   /// Encrypt [bytes].
-  Encrypted encrypt(Uint8List bytes, {IV iv});
+  Encrypted encrypt(Uint8List bytes, {IV? iv});
```

Comment:
`Algorithm` abstract class needs to settle on whether the `IV iv` param is optional or required. This will cascade down into all subclasses.
    * We could subclass `Algorithm` into `AlgorithmWithIV` and that into algorithms that require IV, while those who doesn't just stick to `Algorithm`.

Reply:
I think we can go with the `AlgorithmWithIV` approach.

Reply:
This unfortunately wasn't possible. Instead, I've moved IV where necessary into constructors.